wgpu-native/src/device.rs

```diff
@@ -2112,6 +2113,25 @@ pub extern "C" fn wgpu_device_destroy(device_id: DeviceId) {
     gfx_select!(device_id => device_destroy(device_id))
 }
 
+pub fn device_describe<B: GfxBackend>(device_id: DeviceId) -> RawStringMut {
+    let hub = B::hub();
+    let mut token = Token::root();
+    let (adapter_guard, mut token) = hub.adapters.read(&mut token);
+    let (device_guard, _) = hub.devices.read(&mut token);
+    let device = &device_guard[device_id];
+    let adapter = &adapter_guard[device.adapter_id];
+    let adapter_info = &adapter.raw.info;
+    std::ffi::CString::new(adapter_info.name.clone()).unwrap_or_else(|_|
```

Comment:
It looks like you only output the adapter information. Do you really need the "device" description then?

Reply:
Yep, good point. I was probably interpreting the request in https://github.com/gfx-rs/wgpu-rs/issues/7 too literally. I changed this to be an adapter-level function, which seems more in line with the spirit of the original request.